tracepoints/operational/runtest.sh

```diff
@@ -0,0 +1,232 @@
+
+            stap -DSTP_NO_OVERLOAD $XTRA -t -c "sleep 0.25" ${STAP_VERBOSE_FLAG} group.stap > $VERBOSETRACELOG 2>&1
+            local rc=$?
+            timeCalc
+            if [ $rc -eq 0 ] ; then
+                echo "       Result testing : Test Passed " | tee -a $OUTPUTFILE
+                echo "       Test run time  : $TTIME seconds " | tee -a $OUTPUTFILE
+                echo "------------------------------------------------------------" | tee -a $OUTPUTFILE
+                resultPass $VAR $COUNT
+            else
+                echo "       Result testing : Test Failed " | tee -a $OUTPUTFILE
+                echo "       Test run time  : $TTIME seconds " | tee -a $OUTPUTFILE
+                echo "------------------------------------------------------------" | tee -a $OUTPUTFILE
+                submitLog $VERBOSETRACELOG
+                resultFail $VAR $COUNT
+            fi
+            rm -f $PROBES_FILE
+            rm -f $PROBES_NAME_FILE
+        fi
+    done
+}
+
+function runTest ()
+{
+    local TESTLIST=`/usr/bin/stap -L 'kernel.trace("*")' | grep -o "\".*\""`
+    if [ -z "$TESTLIST" ] ; then
+        resultFail TESTLIST_EMPTY 99
+        exit 0
+    fi
+
+    # Bug 1541287 - WARNING: CPU: 3 PID: 10291 at kernel/jump_label.c:188 __jump_label_update+0x95/0xa0
+    if grep -q "release 7.5" /etc/redhat-release; then
+        TESTLIST=$(echo "$TESTLIST" | sed '/xen:xen_cpu_write_gdt_entry/d')
+        TESTLIST=$(echo "$TESTLIST" | sed '/xen:xen_cpu_write_idt_entry/d')
+    fi
+
+    # Additional argumewnt to stap if Family is RHEL5 and CPU/Family/Model match
+    [ "$FAMILY" == "RedHatEnterpriseLinuxServer5" ] && isCpuFamilyModel AuthenticAMD 21 2
+    if [ "$?" == 0 ]; then
+        XTRA="-DTRYLOCKDELAY=300"
+    fi
+    stap --clean-cache
+    testList "$TESTLIST" 32
+}
+
+# Setup some variables
+if [ -e /etc/redhat-release ] ; then
+    installeddistro=`cat /etc/redhat-release`
+else
+    installeddistro=unknown
+fi
+
+# select tool to manage package, which could be "yum" or "dnf"
+function select_yum_tool() {
+    if [ -x /usr/bin/dnf ]; then
+        echo "/usr/bin/dnf"
+    elif [ -x /usr/bin/yum ]; then
+        echo "/usr/bin/yum"
+    else
+        return 1
+    fi
+
+    return 0
+}
+
+# return 0 when running kernel rt
+is_kernel_rt()
+{
+	local kernel_name=$(uname -r)
+	if [[ "$kernel_name" =~ "rt" ]]; then
+		echo 0
+	else
+		echo 1
+	fi
+}
+
+
+yum=$(select_yum_tool)
+
+kernel=$(uname -r)
+kernbase=$(rpm -q --queryformat '%{name}-%{version}-%{release}.%{arch}\n' -qf /boot/config-$(uname -r))
+stapbase=$(rpm -q --queryformat '%{name}-%{version}-%{release}.%{arch}\n' -qf /usr/bin/stap)
+
+# Checking if running kernel-rt
+if [ $(is_kernel_rt) -eq 0 ]; then
+    ${yum} -y install kernel-rt-devel-${kernel}
+else
+    ${yum} -y install kernel-devel-${kernel}
+fi
+res=$?
+if [ $res -ne 0 ] ; then
+    echo "Failed to install kernel devel-${kernel}" | tee -a $OUTPUTFILE
```

Comment:
What will the exit code be here? Should an explicit warn be reported?

Reply:
adding "report  warn" before aborting.